src/snmalloc/backend/meta_protected_range.h

```diff
@@ -120,6 +120,11 @@ namespace snmalloc
       return meta_range;
     }
 
+    void set_small_heap()
+    {
+      error("Not supported! Cannot use a small heap with MetaProtectedRange.");

```

Comment:
I wonder if we can make this a compile-time error instead... in principle I think we could require `fixedglobalconfig` to do something like:
```
    if constexpr (requires { { local_state->set_small_heap() } -> std::same_as<void> })
```
but that's more `concept`-ual than we've wanted the codebase to be.

Reply:
If I delete this method, then the code still compiles, so we convert it to a compile time error as there is already enough conditional template stuff to make this unreachable.